chromium_src/chrome/browser/ui/bookmarks/bookmark_utils.cc

```diff
@@ -3,13 +3,38 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "chrome/browser/ui/bookmarks/bookmark_utils.h"
+#include "brave/components/constants/pref_names.h"
 #include "build/build_config.h"
 #include "chrome/grit/theme_resources.h"
+#include "components/bookmarks/common/bookmark_pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
 #include "ui/base/resource/resource_bundle.h"
 #include "ui/color/color_id.h"
 #include "ui/color/color_provider.h"
 #include "ui/gfx/color_utils.h"
 
+namespace chrome {
+
+void ToggleBookmarkBarWhenVisible_ChromiumImpl(
+    content::BrowserContext* browser_context);
+
+void BraveToggleBookmarkBarState(content::BrowserContext* browser_context) {
+  auto* prefs = user_prefs::UserPrefs::Get(browser_context);
+  // On macOS with the View menu or via hotkeys, the options Always show
```

Comment:
I still don't understand what is going on here. Why is this based on the current state when this method toggles it? The new state will be `!always_show` after ToggleBookmarkBarWhenVisible_ChromiumImpl is called

Reply:
this call is need just to set kAlwaysShowBookmarkBarOnNTP to false when this function is called

Reply:
the state will be Always or Never acctording to kShowBookmarkBar value

Reply:
@bridiver done, I moved it after `ToggleBookmarkBarWhenVisible_ChromiumImpl`

Reply:
then as discussed in DM, `BookmarkBarSubMenuModel` and `ToggleBookmarkBarWhenVisible` should both call a helper method `SetBookmarkBarState(state, prefs)` so we don't duplicate the logic

Reply:
done